tools/notification-configuration/notification-creator/NotificationConfigurator.cs

```diff
@@ -51,7 +58,7 @@ public NotificationConfigurator(AzureDevOpsService service, ILogger<Notification
                     }
 
                     await EnsureSynchronizedNotificationTeamIsChild(parentTeam, childTeam, persistChanges);
-                    await EnsureScheduledBuildFailSubscriptionExists(pipeline, parentTeam, persistChanges);
+                    // await EnsureScheduledBuildFailSubscriptionExists(pipeline, parentTeam, persistChanges);
```

Comment:
We will still need this but only for newly created teams.

Reply:
Also if we believe getting these subscriptions one at a time is the costly part. They look to have a rest api to get all of them for an org https://docs.microsoft.com/en-us/rest/api/azure/devops/hooks/subscriptions/list?view=azure-devops-rest-6.0, we should consider using that (hopefully there is also a API in the client library we are using for it).

Reply:
It is currently inside of EnsureTeamExists function. We learned where team exists or updates in that function.

Reply:
The link is for service hooks. Is this the one we expect to use?

Reply:
I tried the notification subscriptions apis. They are used to query all types of subscritions: e.g. build.complete, build.failed, Pull request changes

Reply:
So can we not use that to build a cache of all the notifications and map them to the team?

>It is currently inside of EnsureTeamExists function

OK yes I see you moved it there.